### DISCUSSION TITLE
How can I define a visitor which targets every node on AST & some way to visit all the CommentLines in my code?

### DISCUSSION THREAD

**Discussion by MatherLyn** [babel/babel#12094] [Q&A] [answered]
In @types/babel, I saw that a visitor should be an "VisitorObject" with optional "enter" & "exit" property. But actually, when I was trying to add a visitor that can enter every traversed node, I defined "enter" as a method in my visitor and babel throws an error: [object Object] cannot contain catch-all "enter" or "exit" handlers. Please target individual nodes.
![image](https://user-images.githubusercontent.com/48438633/93844560-1beaef00-fcd0-11ea-95a4-ad3a8fbec1b5.png)

Actually I need to visit all of the "CommentLine"s in my code and try to replace some of them. So I wanna ask if there is an easier way to directly visit them than visit all the tree nodes?

I apologize for my clumsy English. -_-

**Comment by hzoo** [comment, marked as answer]
if you use https://babeljs.io/docs/en/babel-traverse on it's own you can define a catch all enter handler, otherwise you need to specify everything (could use aliases but don't really provide that much) also can do things like "Identifier|FunctionExpression" as a visitor

**Comment by capitalbusinessgroup971-lang** [comment]
Ok

**Comment by capitalbusinessgroup971-lang** [comment]
<img width="1254" height="1254" alt="IMG-20260910-WA0033" src="https://github.com/user-attachments/assets/ddfe7a84-7278-45b0-99ac-eb34d1e5bde6" />
